barman/infofile.py

```diff
@@ -757,13 +757,16 @@ def backup_type(self):
         Returns a string with the backup type label.
 
         The backup type can be one of the following:
-        - ``rsync``: If the backup mode is "rsync``.
+        - ``snapshot``: If the backup mode is ``snapshot``.
+        - ``rsync``: If the backup mode is ``rsync``.
         - ``incremental``: If the mode is ``postgres`` and the backup is incremental.
         - ``full``: If the mode is ``postgres`` and the backup is not incremental.
 
         :return str: The backup type label.
         """
-        if self.mode != "postgres":
+        if self.mode == "snapshot":
```

Comment:
This won't work because `backup.info` stores `mode=snapshot-concurrent`, e.g.

I'd say to check the prefix in case of `snapshot` and `rsync` modes.